### DISCUSSION TITLE
Supporting tooltips on Area glyphs

### DISCUSSION THREAD

**Discussion by xaptronic** [bokeh/bokeh#11979] [Internals and design (Q&A)]
cc: @bokeh/dev

Hello! I've gone through the hit-testing / index selection code related to Areas / Glyphs and come up with a way to get this to work.

The core of the change is that, instead of doing hit testing on the entire area glyph polygon, I changed the implementation to iterate throug each segment of the Polygon (essentially "sub-polygons") and then if hit test passes, mark the index for use by the `_update` method in `hover_tooltip.ts`.

I have [posted a proof-of-concept of this with working examples](https://github.com/xaptronic/bokeh/pull/2/files) on my bokeh fork, and would welcome some feedback in terms of direction of this change.

The VAreaView handling code in `hover_tooltip.ts` is especially proof-of-concept - really just some hacking to get it work.

Thanks!

**Comment by bryevdv** [comment]
> The core of the change is that, instead of doing hit testing on the entire area glyph polygon, I changed the implementation to iterate throug each segment of the Polygon 

@xaptronic I am not sure this is advised? (or maybe I am misunderstanding somethign) For tooltips we would definitely want to pop up a hover any time the mouse is inside the (one, single) polygon that defines the `VArea`. That implies one point in poly hit test for the one poygon. I don't know what sub-polygons means in this context. Each `VArea` only defines one single polygon. Looking at the code, I actually think perhaps the existing hit-testing code was fine, but all it did was set the `result.view = this` but the hover tool does not know how to do anything with that information. I think the work might really just be to teach the hover tool how to handle a hit on a polygon that does not report hits by setting *indices*

**Reply by bryevdv** [reply]
I guess I would add: It's possible that we might *also* want to report `line_indices`, maybe there are cases when it is useful to know what what specific sub-segments of the `VArea` were hit, but that would not be what drives the hover tool. E.g. the hover tool should pop up when the mouse is in the *middle* of a `VArea` in which case no line segments are "hit" at all. But also reporting line indices for "hits" on sub-segments of the one `VArea` would require using a real point-segment hit test function on each sub-segment. I don't think reporting line indices is a priority for getting he hover tool hooked up.
